reana_workflow_controller/rest.py

```diff
@@ -147,6 +148,79 @@ def get_workflows():  # noqa
         return jsonify({"message": str(e)}), 500
 
 
+@restapi_blueprint.route('/workflows/<workflow_id>/workspace',
+                         methods=['POST'])
+def seed_workflow_workspace(workflow_id):
+    r"""Seed workflow workspace.
+
+    ---
+    post:
+      summary: Adds a file to the workflow workspace.
+      description: >-
+        This resource is expecting a workflow UUID and a file to place in the
+        workflow workspace.
+      operationId: seed_workflow
+      consumes:
+        - multipart/form-data
+      produces:
+        - application/json
+      parameters:
+        - name: organization
+          in: query
+          description: Required. Organization which the worklow belongs to.
+          required: true
+          type: string
+        - name: user
+          in: query
+          description: Required. UUID of workflow owner.
+          required: true
+          type: string
+        - name: workflow_id
+          in: path
+          description: Required. Workflow UUID.
+          required: true
+          type: string
+        - name: file_content
+          in: formData
+          description: Required. File to add to the workflow workspace.
+          required: true
+          type: file
+        - name: file_name
+          in: query
+          description: Required. File name.
+          required: true
+          type: string
+      responses:
+        200:
+          description: >-
+            Request succeeded. The file has been added to the workspace.
+          schema:
+            type: object
+            properties:
+              message:
+                type: string
+              workflow_id:
+                type: string
+          examples:
+            application/json:
+              {
+                "message": "Workflow has been added to the workspace",
+                "workflow_id": "cdcf48b1-c2f3-4693-8230-b066e088c6ac"
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed
+    """
+    try:
+        file_ = request.files['file_content'].stream.read()
+        file_name = secure_filename(request.args['file_name'])
```

Comment:
Adding this here just as  a reminder:
> The function might return an empty filename. It’s your responsibility to ensure that the filename is unique and that you generate random filename if the function returned an empty one.
(http://werkzeug.pocoo.org/docs/0.12/utils/#werkzeug.utils.secure_filename)

Reply:
thanks 😄 , linked to the workspace creation issue (when we will write files to disk)